neo/Consensus/ConsensusService.cs

```diff
@@ -268,7 +272,18 @@ private void OnConsensusPayload(ConsensusPayload payload)
         private void OnPersistCompleted(Block block)
         {
             Log($"persist block: {block.Hash}");
-            block_received_time = TimeProvider.Current.UtcNow;
+            reference_block_time = TimeProvider.Current.UtcNow;        
+            theoreticalDelay = reference_block_time.ToTimestamp() - block.Timestamp;
```

Comment:
Why not just put an I around this whole code section so it only runs if `reference_block_time.ToTimestamp() > block.Timestamp` Instead of checking if < 0 at the end.

Reply:
> Maybe Peeeter @PeterLinX could give us a hand here when possible.

+1

Reply:
I agree.

Reply:
Check it know, I also made it to calculate inside a function. Looks clear now.

After experiments we can clear more if we decide to merge it.